src/internal/cache-client.ts

```diff
@@ -716,6 +743,81 @@ export class CacheClient {
     });
   }
 
+  public async listRetain(
+    cacheName: string,
+    listName: string,
+    startIndex?: number,
+    endIndex?: number,
+    ttl: CollectionTtl = CollectionTtl.fromCacheTtl()
+  ): Promise<CacheListRetain.Response> {
+    try {
+      validateCacheName(cacheName);
+      validateListName(listName);
+    } catch (err) {
+      return new CacheListRetain.Error(normalizeSdkError(err as Error));
+    }
+    this.logger.trace(
+      `Issuing 'listRetain' request;
+      listName: ${listName},
+      startIndex: ${startIndex ?? 'null'},
+      endIndex: ${endIndex ?? 'null'},
+      ${ttl.toString()}`
```

Comment:
Can you use a format string instead of interpolating the values? That is a pattern we've adopted elsewhere to avoid the extra compute when the level isn't trace.